lib/standard/bytes.nit

```diff
@@ -0,0 +1,151 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Services for byte streams and arrays
+module bytes
+
+import kernel
+import collection::array
+intrude import string
+
+# A buffer containing Byte-manipulation facilities
+#
+# TODO: Change the bound to Byte when available in stdlib and bootstrap
+class ByteBuffer
+	super AbstractArray[Int]
+
+	# A NativeString being a char*, it can be used as underlying representation here.
+	private var items: NativeString
+
+	# Number of bytes in the array
+	redef var length: Int
+
+	# Capacity of the array
+	private var capacity: Int
+
+	# Has this buffer been persisted (to_s'd)?
+	private var persisted = false
+
+	init empty do
+		var ns = new NativeString(0)
+		init(ns, 0, 0)
+	end
+
+	init with_capacity(cap: Int) do
+		var ns = new NativeString(cap)
+		init(ns, 0, cap)
+	end
+
+	redef fun is_empty do return length != 0
+
+	redef fun [](i: Int): Int do
+		assert i >= 0
+		assert i < length
+		return items[i].ascii
+	end
+
+	redef fun []=(i: Int, v: Int) do
+		if persisted then regen
+		assert i >= 0
+		assert i < length
+		items[i] = v.ascii
+	end
+
+	redef fun add(c: Int) do
+		if persisted then regen
+		if length >= capacity then
+			enlarge(length)
+		end
+		items[length] = c.ascii
+		length += 1
+	end
+
+	redef fun append(arr: Collection[Int]) do
+		if arr isa ByteBuffer then
+			append_ns(arr.items, arr.length)
+		else
+			for i in arr do add i
+		end
+	end
+
+	redef fun clear do length = 0
+
+	# Regenerates the buffer, necessary when it was persisted
+	private fun regen do
```

Comment:
then you should say it somewhere. in the public doc of the class at least since it is a behavioral feature. Maybe also in the doc of this method and/or the doc of `persisted`